perllib/FixMyStreet/App/Form/Waste/Garden.pm

```diff
@@ -110,7 +110,11 @@ has_page summary => (
         $data->{cost_pa} = $cost_pa / 100;
         $data->{display_total} = $total / 100;
 
-        return {};
+        my $button_text = 'Continue to payment';
+        if ($form->{c}->cobrand->feature('waste_features') && $form->{c}->cobrand->feature('waste_features')->{text_for_waste_payment}) {
```

Comment:
I think this doesn't need the first half? feature() will return undef and you can look up a key on that without warning (unless I've missed something).

Reply:
Running test suite getting errors like '[error] Caught exception in FixMyStreet::App::Controller::Waste->form "Can't use an undefined value as a HASH reference at /home/moray/FixMyStreet/fixmystreet/perllib/FixMyStreet/App/Form/Waste/Garden/Renew.pm line 80."' so thought that was the issue

Reply:
Fair enough, then!

Reply:
Though I don't see why it wouldn't: ` perl -we'use strict; my $d; my $e = $d->{123};` doesn't error.

Reply:
Ah, so it's okay if you put it in a variable first, but not if you try and use it directly.